include/swift/AST/ASTContext.h

```diff
@@ -411,7 +412,11 @@ class ASTContext final {
 
   /// Cache of module names that passed the 'canImport' test. This cannot be
   /// mutable since it needs to be queried for dependency discovery.
-  llvm::StringSet<> SucceededModuleImportNames;
+  struct ImportedModuleVersionInfo {
```

Comment:
Nit: Doc comment should be on the map; why is it std::map instead of StringMap (or MapVector around StringMap if you need a stable order)? Does it really need to be sorted by key?

Reply:
It is either this needs to be sorted or the scanner that request the module version needs to sort it afterwards before converting to the build args. Since options do not get canonicalized now, the order of the version matters.